starlite/stores/registry.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+if TYPE_CHECKING:
+    from .base import Store
+
+
+from .memory import MemoryStore
+
+__all__ = ("StoreRegistry",)
+
+
+def default_default_factory(name: str) -> Store:
+    return MemoryStore()
+
+
+class StoreRegistry:
```

Comment:
I don't see why this class is required, can you elaborate?

Reply:
I did. Read the initial comment on the PR. This specifically addresses several issues you raised.